contrib/drivers/scutech/cms/replication.go

```diff
@@ -0,0 +1,127 @@
+// limitations under the License.
+
+package scms
+
+import (
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"github.com/opensds/opensds/pkg/model"
+	pb "github.com/opensds/opensds/pkg/model/proto"
+)
+
+// RelicationDriver
+type ReplicationDriver struct{}
+
+// Setup
+func (r *ReplicationDriver) Setup() error { return nil }
+
+// Unset
+func (r *ReplicationDriver) Unset() error { return nil }
+
+// Create and start replication
+func (r *ReplicationDriver) CreateReplication(opt *pb.CreateReplicationOpts) (*model.ReplicationSpec, error) {
+	glog.Infoln("CMS create migration task ...")
+
+	replica := &model.ReplicationSpec{
+		// TODO: return additional important information
+		PrimaryReplicationDriverData:   make(map[string]string),
+		SecondaryReplicationDriverData: make(map[string]string),
+	}
+
+	isPrimary := opt.GetIsPrimary()
+	if !isPrimary {
+                //on CMSProxy, do nothing
+		return replica, nil
+	}
+
+	bandwidth := opt.GetReplicationBandwidth()
+	primaryData := opt.GetPrimaryReplicationDriverData()
+	secondaryData := opt.GetSecondaryReplicationDriverData()
+	primaryVolId := opt.GetPrimaryVolumeId()
+	secondaryVolId := opt.GetSecondaryVolumeId()
+	path, _ := filepath.EvalSymlinks(primaryData["Mountpoint"])
+	primaryBackingDevice, _ := filepath.Abs(path)
+
+	path, _ = secondaryData["Mountpoint"]
+	secondaryBackingDevice, _ := filepath.Abs(path)
+	glog.Infof("%s:%s\n", primaryBackingDevice, secondaryBackingDevice)
+	sourceVol := CmsVolume{VolumeId: primaryVolId, VolumeName: primaryBackingDevice}
+	targetVol := CmsVolume{VolumeId: secondaryVolId, VolumeName: secondaryBackingDevice}
+
+	task := NewCmsTask(bandwidth, false)
+	if err := task.AddVolume(sourceVol, targetVol); err != nil {
+		return nil, err
+	}
+
+	cmsadm := NewCmsAdm()
+	if _, err := cmsadm.CreateTask(task); err != nil {
+		return nil, err
+	}
+
+	if _, err := cmsadm.Up(); err != nil {
+                return nil, err
+        }
+
+	return replica, nil
+}
+
+//delete replication
+func (r *ReplicationDriver) DeleteReplication(opt *pb.DeleteReplicationOpts) error {
+	glog.Infoln("CMS delete migration task ...")
+
+	isPrimary := opt.GetIsPrimary()
+	if !isPrimary {
+		return nil
+	}
+
+	cmsadm := NewCmsAdm()
+	_, err := cmsadm.DeleteTask()
+	return err
+}
+
+//start replication
+func (r *ReplicationDriver) EnableReplication(opt *pb.EnableReplicationOpts) error {
+	glog.Infoln("CMS start migration task ....")
+
+	isPrimary := opt.GetIsPrimary()
+	if !isPrimary {
+		return nil
+	}
+
+	cmsadm := NewCmsAdm()
+	_, err := cmsadm.Up()
+	return err
+}
+
+//stop replication
+func (r *ReplicationDriver) DisableReplication(opt *pb.DisableReplicationOpts) error {
+	glog.Infoln("CMS stop migration task")
+
+	isPrimary := opt.GetIsPrimary()
+	if !isPrimary {
+		return nil
+	}
+
+	cmsadm := NewCmsAdm()
+
+	_, err := cmsadm.Down()
+	return err
+}
+
+//
+func (r *ReplicationDriver) FailoverReplication(opt *pb.FailoverReplicationOpts) error {
+	glog.Infoln("CMS failover ....")
+
```

Comment:
Please add some comments explaining why this function does not do anything.  See example comments in DRBD driver:

https://github.com/opensds/opensds/blob/master/contrib/drivers/drbd/replication.go#L235